x-pack/plugins/infra/public/containers/logs/with_log_filter.tsx

```diff
@@ -33,3 +35,39 @@ export const withLogFilter = connect(
 );
 
 export const WithLogFilter = asChildFunctionRenderer(withLogFilter);
+
+/**
+ * Url State
+ */
+
+type LogFilterUrlState = ReturnType<typeof logFilterSelectors.selectLogFilterQuery>;
+
+export const WithLogFilterUrlState = () => (
+  <WithLogFilter>
+    {({ applyFilterQuery, filterQuery }) => (
+      <UrlStateContainer<LogFilterUrlState>
```

Comment:
The mixing TS generics and JSX together reads weird. Can you assign that to a variable then just use the variable? I haven't tried that before so I'm not sure if it will work but it if it does it might make it more readable.

Reply:
Unfortunately that is no easily possible. The generic type argument has to be passed at instantiation time, which is handled by JSX. If I would convert the `UrlStateContainer` component into a class-based component, a no-op inheritance could nail down the type. Alternatively, we could manually type the passed properties sufficiently so the inference figures it out. Typing the return type of `mapToUrlState` should do it. Would you think that improves the situation?

```typescript
<UrlStateContainer
  urlState={urlState}
  urlStateKey="logPosition"
  mapToUrlState={mapToUrlState}
  ...
/>

const mapToUrlState = (value: any): LogPositionUrlState | undefined => ...
```

Reply:
ok, change coming up...